operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/event/EventSourceManagerTest.java

```diff
@@ -85,12 +85,12 @@ public void startCascadesToEventSources() {
 
   @Test
   void retrievingEventSourceForClassShouldWork() {
-    assertThatExceptionOfType(IllegalArgumentException.class)
+    assertThatExceptionOfType(NoEventSourceForClassException.class)
         .isThrownBy(() -> eventSourceManager.getEventSourceFor(Class.class));
 
     // manager is initialized with a controller configured to handle HasMetadata
     EventSourceManager manager = initManager();
-    assertThatExceptionOfType(IllegalArgumentException.class)
+    assertThatExceptionOfType(NoEventSourceForClassException.class)
```

Comment:
This will require an entry in the upgrade notes

Reply:
Will do!